src/pydata_sphinx_theme/toctree.py

```diff
@@ -459,36 +474,54 @@ def add_collapse_checkboxes(soup: BeautifulSoup) -> None:
         element.insert(1, checkbox)
 
 
-def get_local_toctree_for_doc(
-    toctree: TocTree, indexname: str, pagename: str, builder, collapse: bool, **kwargs
-) -> List[BeautifulSoup]:
-    """Get the "local" TocTree containing `pagename` rooted at `indexname`.
-
-    The Sphinx equivalent is TocTree.get_toctree_for(), which always uses the "root"
-    or "global" TocTree:
-
-        doctree = self.env.get_doctree(self.env.config.root_doc)
-
-    Whereas here we return a subset of the global toctree, rooted at `indexname`
-    (e.g. starting at a second level for the sidebar).
+def get_nonroot_toctree(
+    app: Sphinx, pagename: str, ancestorname: str, toctree, **kwargs
+):
+    """Get the partial TocTree (rooted at `ancestorname`) that dominates `pagename`.
+
+    Parameters:
+    app : Sphinx app.
+    pagename : Name of the current page (as Sphinx knows it; i.e., its relative path
+    from the documentation root).
+    ancestorname : Name of a page that dominates `pagename` and that will serve as the
+    root of the TocTree fragment.
+    toctree : A Sphinx TocTree object. Since this is always needed when finding the
+    ancestorname (see _get_ancestor_pagename), it's more efficient to pass it here to
+    re-use it.
+    kwargs : passed to the Sphinx `toctree` template function.
+
+    This is similar to `context["toctree"](**kwargs)` (AKA `toctree(**kwargs)` within a
+    Jinja template), or `TocTree.get_toctree_for()`, which always uses the "root"
+    doctree (i.e., `doctree = self.env.get_doctree(self.env.config.root_doc)`).
     """
-    partial_doctree = toctree.env.tocs[indexname].deepcopy()
-
-    toctrees = []
+    kwargs.setdefault("collapse", True)
     if "maxdepth" not in kwargs or not kwargs["maxdepth"]:
         kwargs["maxdepth"] = 0
     kwargs["maxdepth"] = int(kwargs["maxdepth"])
-    kwargs["collapse"] = collapse
-
-    # TODO: use `doctree.findall(TocTreeNodeClass)` once docutils min version >=0.18.1
-    for _node in traverse_or_findall(partial_doctree, TocTreeNodeClass):
-        # defaults for resolve: prune=True, maxdepth=0, titles_only=False, collapse=False, includehidden=False
-        _toctree = toctree.resolve(pagename, builder, _node, **kwargs)
-        if _toctree:
-            toctrees.append(_toctree)
+    # starting from ancestor page, recursively parse `toctree::` elements
+    ancestor_doctree = toctree.env.tocs[ancestorname].deepcopy()
+    toctrees = []
+
+    # for each `toctree::` directive in the ancestor page...
+    for toctree_node in traverse_or_findall(ancestor_doctree, TocTreeNodeClass):
+        # TODO: ↑↑↑↑↑↑ use `ancestor_doctree.findall(TocTreeNodeClass)` ↑↑↑↑↑↑
+        #              once docutils min version >=0.18.1
```

Comment:
```suggestion
        #              once docutils min version >=0.18.1 (sphinx >=7)
```

Reply:
Where did you find the info that sphinx 7 is dropping support for docutils 0.18.1? From what I see [here](https://www.sphinx-doc.org/en/master/changes.html#release-7-0-1-released-may-12-2023) sphinx 7.0.1 *adds* support for docutils 0.20 but there is no mention there of *dropping* support for older versions. The most recent version bump mentioned is that in [sphinx 6.2.0](https://www.sphinx-doc.org/en/master/changes.html#release-6-2-0-released-apr-23-2023) they dropped support for anything *older* than docutils 0.18.1.

FWIW if we were to take inspiration from [SPEC0](https://scientific-python.org/specs/spec-0000/) then we would be OK to drop support for docutils 0.18.1 now (as of [2023-12-23](https://docutils.sourceforge.io/RELEASE-NOTES.html#release-0-18-1-2021-12-23)) as it is now more than 2 years since its release and there are 3 versions newer than it.  On the contrary, SPEC0 would suggest that we support Sphinx 4.4.0 up until yesterday, and support sphinx 4.5.0 through March 2024 (which is more than we currently do)

Reply:
I checked the pyproject.toml file from Sphinx: https://github.com/sphinx-doc/sphinx/blame/master/pyproject.toml

the current version (>7) is setting docutils to `docutils>=0.18.1,<0.21`. 
using the blame display I checked when the lower bounds was set (https://github.com/sphinx-doc/sphinx/commit/ae9008b1282278ae9ab7e53fa0d87e1a7be19e12) which was released in v6.2.0. So I made a mistake.